addons/godot-xr-tools/user_settings/user_settings.gd

```diff
@@ -22,19 +46,42 @@ func reset_to_defaults() -> void:
 	# Reset to defaults
 	snap_turning = XRTools.get_default_snap_turning()
 	player_height_adjust = 0.0
+	webxr_primary = WebXRPrimary.AUTO
+	webxr_auto_primary = 0
 
 
 ## Set the player height adjust property
 func set_player_height_adjust(new_value : float) -> void:
 	player_height_adjust = clamp(new_value, -1.0, 1.0)
 
 
+## Set the WebXR primary
+func set_webxr_primary(new_value : int) -> void:
+	webxr_primary = new_value
+	if webxr_primary == WebXRPrimary.AUTO:
+		if webxr_auto_primary == 0:
+			# Don't emit the signal yet, wait until we detect which to use.
+			pass
+		else:
+			webxr_primary_changed.emit(webxr_auto_primary)
+	else:
+		webxr_primary_changed.emit(webxr_primary)
+
+
+## Gets the WebXR primary (taking into account auto detection).
+func get_real_webxr_primary() -> int:
+	if webxr_primary == WebXRPrimary.AUTO:
+		return webxr_auto_primary
+	return webxr_primary
+
+
 ## Save the settings to file
 func save() -> void:
 	# Construct the settings dictionary
 	var data = {
 		"input" : {
-			"default_snap_turning" : snap_turning
+			"default_snap_turning" : snap_turning,
```

Comment:
I would create a root node "webxr" and place all webxr settings in there instead of in "input"

Reply:
Done in latest push!